includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1543,15 +1564,19 @@ public function save_intent_to_order( $order, $intent ) {
 	public function get_intent_from_order( $order ) {
 		$intent_id = $order->get_meta( '_stripe_intent_id' );
 
-		if ( $intent_id ) {
-			return $this->get_intent( 'payment_intents', $intent_id );
-		}
+		try {
+			if ( $intent_id ) {
+				return $this->get_intent( 'payment_intents', $intent_id );
+			}
 
-		// The order doesn't have a payment intent, but it may have a setup intent.
-		$intent_id = $order->get_meta( '_stripe_setup_intent' );
+			// The order doesn't have a payment intent, but it may have a setup intent.
+			$intent_id = $order->get_meta( '_stripe_setup_intent' );
 
-		if ( $intent_id ) {
-			return $this->get_intent( 'setup_intents', $intent_id );
+			if ( $intent_id ) {
+				return $this->get_intent( 'setup_intents', $intent_id );
+			}
+		} catch ( WC_Stripe_Exception $e ) {
```

Comment:
For `get_intent_from_order`, the following methods expect the exception:
- WC_Gateway_Stripe::process_payment *
- WC_Gateway_Stripe::prepare_order_pay_page * → prepare_intent_for_order_pay_page
- WC_Gateway_Stripe::render_payment_intent_inputs * → prepare_intent_for_order_pay_page